src/main/java/io/seqera/tower/cli/responses/runs/RunViewMetrics.java

```diff
@@ -67,7 +75,90 @@ public Object getJSON() {
         return data;
     }
 
-    private void processDataTable(List<Map<String, Object>> metricData, PrintWriter out, List<String> cols) {
+    @Override
+    public void toString(PrintWriter out) {
+        List<String> fields = columns.stream().map(Enum::name).collect(Collectors.toList());
+        List<String> cols = new ArrayList<>();
+        cols.add("process");
+        if (groupType == MetricPreviewFormat.expanded) {
+            cols.add("metric");
+        }
+        cols.addAll(fields);
+
+        if (!metricsMem.isEmpty()) {
+            out.println(ansi(String.format("%n%n    @|bold  Memory Metrics|@%n    ----------------%n")));
+
+            Map<String, Function<Float, Object>> formatter = new HashMap<>();
+            formatter.put("memVirtual", FormatHelper::formatBits);
+            formatter.put("memRaw", FormatHelper::formatBits);
+            formatter.put("memUsage", FormatHelper::formatPercentage);
+
+            if (groupType == MetricPreviewFormat.condensed) {
+                out.println(ansi(String.format("   @|italic   Legend:  physical RAM / virtual RAM+swap / %%RAM allocated |@%n")));
+                processDataReducedTable(metricsMem, out, cols, formatter);
+            } else {
+                processExpandedDataTable(metricsMem, out, cols, formatter);
+            }
+        }
+
+
+        if (!metricsCpu.isEmpty()) {
+            out.println(ansi(String.format("%n%n    @|bold  CPU Metrics|@%n    ----------------%n")));
+
+            Map<String, Function<Float, Object>> formatter = new HashMap<>();
```

Comment:
Why not create only one static formatter with all the formatters?  You can create a static map or a helper class `formatters.get(String sectionName) -> Function<Number, Object>`